### PR TITLE
added database connection

### PR DESCRIPTION
**Pull request by lilymammadova**

### CODE REVIEW

File: .idea/artifacts/HotelApp_war_exploded.xml
Comment (Alexmansar) [inline .idea/artifacts/HotelApp_war_exploded.xml:1]: remove from git

File: .idea/dataSources.xml
Comment (Alexmansar) [inline .idea/dataSources.xml:1]: the same

File: src/main/java/org/liliya/hotelapp/configuration/Configuration.java [hunk 1]
```diff
@@ -5,33 +5,52 @@
...
         loadProperties();
     }
 
+    public static Configuration getInstance() {
+        if (instance == null) {
+            instance = new Configuration();
+        }
+        return instance;
+    }
+
```
Comment (Alexmansar) [inline src/main/java/org/liliya/hotelapp/configuration/Configuration.java:24]: Singlton ?
👍

File: src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java [hunk 2]
```diff
@@ -1,92 +1,202 @@
...
+        this.configuration = Configuration.getInstance();
+        this.databaseConnection = databaseConnection;
     }
 
     @Override
-    public void register(double price) {
-        Apartment apartment = new Apartment(idCounter++, price, ReservationStatus.AVAILABLE, null);
-        apartments.add(apartment);
+    public int register(double price) {
+        String query = "INSERT INTO apartments (price, status) VALUES (?, ?)";
```
Comment (Alexmansar) [inline src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java]: use all queries like a constants in the class level

File: src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java [hunk 3]
```diff
@@ -1,92 +1,202 @@
...
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return generatedId;
     }
 
     @Override
     public boolean reserve(Client client) {
```
Comment (Alexmansar) [inline src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java:66]: in general split method with small request.
1) insertClient
2)findAvailable
3) updateApartmen
and use just here))

File: src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java [hunk 4]
```diff
@@ -1,92 +1,202 @@
...
-            return apartments.stream().filter(apartment -> apartment.getId() == id)
-                    .findFirst()
-                    .map(apartment -> {
-                        apartment.setReservationStatus(ReservationStatus.AVAILABLE);
-                        apartment.setClient(null);
-                        return true;
-                    })
-                    .orElse(false);
+        if (!configuration.statusChangeAvailability()) {
+            return false;
```
Comment (Alexmansar) [inline src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java:98]: same

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@lilymammadova Hi) I have some comments. Please, check and fix them)

File: src/main/java/org/liliya/hotelapp/configuration/Configuration.java (hunk 1 above)
Comment (lilymammadova) [reply src/main/java/org/liliya/hotelapp/configuration/Configuration.java:24]: yes, I used it for creating configuration instance once

**Review by Alexmansar — APPROVED** [review]
@lilymammadova Please, merge it))

**Pull request merged**